staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -331,43 +416,73 @@ type EventCorrelateResult struct {
 	Skip bool
 }
 
-// NewEventCorrelator returns an EventCorrelator configured with default values.
-//
-// The EventCorrelator is responsible for event filtering, aggregating, and counting
-// prior to interacting with the API server to record the event.
-//
-// The default behavior is as follows:
-//   * No events are filtered from being recorded
+// Spam limit and aggregation parameters for EventCorrelator
+type EventCorrelatorOptions struct {
+	// The maximum amount of lru cache entries for the aggregation state
+	CacheSize int
+	// The amount of events allowed per source and object
+	SpamBurst int
+	// Refill rate of the token bucket in queries per second
+	SpamQPS float32
+	// The maximum number of events in the specified interval before aggregation occurs
+	AggregateMaxEvents int
+	// The amount of time in seconds that must transpire since the last occurrence of a similar event before
+	// it's considered new
+	AggregateIntervalInSeconds int
+}
+
+// NewDefaultEventCorrelatorOptions returns an option set containing the default EventCorrelator behavior:
 //   * Aggregation is performed if a similar event is recorded 10 times in a
 //     in a 10 minute rolling interval.  A similar event is an event that varies only by
 //     the Event.Message field.  Rather than recording the precise event, aggregation
 //     will create a new event whose message reports that it has combined events with
 //     the same reason.
 //   * Events are incrementally counted if the exact same event is encountered multiple
 //     times.
-func NewEventCorrelator(clock clock.Clock) *EventCorrelator {
-	cacheSize := maxLruCacheEntries
+//   * A source may burst 25 events about an object, but has a refill rate budget
+//     per object of 1 event every 5 minutes to control long-tail of spam.
+func NewDefaultEventCorrelatorOptions() *EventCorrelatorOptions {
+	return NewEventCorrelatorOptions(maxLruCacheEntries, defaultSpamBurst, defaultSpamQPS, defaultAggregateMaxEvents, defaultAggregateIntervalInSeconds)
+}
+
+func NewEventCorrelatorOptions(cacheSize, spamBurst int, spamQPS float32, aggregateMax, aggregateInterval int) *EventCorrelatorOptions {
```

Comment:
This function seems not useful and will have to change every time we add a filed to EventCorrelatorOptions